src/mainwindow/mainwindow.cpp

```diff
@@ -63,31 +67,170 @@ MainWindow::~MainWindow()
 // SLOTS
 ///////////////////////////////////////////////////////////
 
+void MainWindow::on_updateWidget_clicked()
```

Comment:
Let's avoid confusion and make it clear this is only for testing, change this function to onTestUpdateWidget_clicked and update the button in the .ui file accordingly.